servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/ConnectionAcceptingNettyHttpServerTest.java

```diff
@@ -0,0 +1,131 @@
+    private static final int TCP_BACKLOG = IS_LINUX ? 0 : 1;
+    private static final int CONNECT_TIMEOUT_MILLIS = (int) ofSeconds(5).toMillis();
+    private static final int VERIFY_REQUEST_AWAIT_SECS = 5;
+    private static final int TRY_REQUEST_AWAIT_SECS = 1;
+
+    private final StreamingHttpRequestResponseFactory reqRespFactory =
+            new DefaultStreamingHttpRequestResponseFactory(DEFAULT_ALLOCATOR, INSTANCE, HTTP_1_1);
+
+    @Override
+    protected void configureServerBuilder(final HttpServerBuilder serverBuilder) {
+        serverBuilder.listenSocketOption(ServiceTalkSocketOptions.SO_BACKLOG, TCP_BACKLOG);
+    }
+
+    @Test
+    void testStopAcceptingAndResume() throws Exception {
+        setUp(CACHED, CACHED);
+        final StreamingHttpRequest request = reqRespFactory.newRequest(GET, SVC_ECHO);
+
+        assertConnectionRequestSucceeds(request);
+
+        serverContext().acceptConnections(false);
+        // Netty will evaluate the auto-read on the next round, so the next connection will go through.
+        assertConnectionRequestSucceeds(request);
+        // This connection should get established but not accepted.
+        assertConnectionRequestReceiveTimesOut(request);
+
+        // Restoring auto-read will resume accepting.
+        serverContext().acceptConnections(true);
+        assertConnectionRequestSucceeds(request);
+    }
+
+    @Test
+    void testIdleTimeout() throws Exception {
+        setUp(CACHED, CACHED);
+        final StreamingHttpRequest request = reqRespFactory.newRequest(GET, SVC_ECHO);
+        assertConnectionRequestSucceeds(request);
+
+        serverContext().acceptConnections(false);
+        // Netty will evaluate the auto-read on the next round, so the next connection will go through.
+        assertConnectionRequestSucceeds(request);
+        // Connection will establish but remain in the accept-queue
+        // (i.e., NOT accepted by the server => occupying 1 backlog entry)
+        assertConnectionRequestReceiveTimesOut(request);
+        final StreamingHttpClient httpClient2 = newClient();
+        // Since we control the backlog size, this connection won't establish (i.e., NO syn-ack)
+        // timeout operator can be used to kill it or socket connection-timeout
+        final ExecutionException executionException =
+                assertThrows(ExecutionException.class, () -> awaitIndefinitely(httpClient2.request(request)));
+        assertThat(executionException.getCause(), instanceOf(ConnectTimeoutException.class));
+    }
+
+    private void assertConnectionRequestReceiveTimesOut(final StreamingHttpRequest request) {
+        final StreamingHttpClient httpClient = newClient();
+        assertThrows(TimeoutException.class, () -> await(httpClient.request(request), TRY_REQUEST_AWAIT_SECS, SECONDS));
+    }
+
+    private StreamingHttpClient newClient() {
+        return newClientBuilder()
+                .appendConnectionFactoryFilter(withMax(1))
```

Comment:
We are leaking resources if we create new clients and never close them. `AbstractNettyHttpServerTest` already has a client. Instead of creating new clients, you can take the existing one using `streamingHttpClient()` and use its API to open new connections when you need them: `client.reserveConnection(...)`. That way, `AbstractNettyHttpServerTest` will take care of properly closing all resouces when a test completes.

To test the difference between `toFuture().get(...)` timeout and `CONNECT_TIMEOUT_MILLIS`, just use lower duration for `toFuture().get(...)`: half a sec vs second.